src/core/bonded_interactions/angle_common.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+  Copyright (C) 2010-2019 The ESPResSo project
+  Copyright (C) 2002-2010
+    Max-Planck-Institute for Polymer Research, Theory Group
+
+  This file is part of ESPResSo.
+
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef ANGLE_COMMON_H
+#define ANGLE_COMMON_H
+/** \file
+ *  Common code for functions calculating angle forces.
+ */
+
+#include "grid.hpp"
+#include "utils.hpp"
+
+/** Compute the normalized vector between two points.
+ *  @param[in]  r_mid     Position of the middle particle
+ *  @param[in]  r_out     Position of the outer particle
+ *  @param[out] vec       Normalized vector
+ *  @param[out] di        Normalization constant
+ */
+inline void calc_angle_vector(const Vector3d &r_mid, const Vector3d &r_out,
+                              double vec[3], double &di) {
+  /* vector from r_out to r_mid */
+  get_mi_vector(vec, r_mid, r_out);
+  di = 1.0 / sqrt(sqrlen(vec));
+  for (int j = 0; j < 3; j++)
+    vec[j] *= di;
+}
+
+/** Compute the angle interaction force.
+ *  @param[out] force1    Force on particle 1
+ *  @param[out] force2    Force on particle 2
+ *  @param[in]  vec1      Normalized vector for particle 1
+ *  @param[in]  vec2      Normalized vector for particle 2
+ *  @param[in]  d1i       Normalization constant for @p vec1
+ *  @param[in]  d2i       Normalization constant for @p vec2
+ *  @param[in]  cosine    Cosine of the angle
+ *  @param[in]  fac       Angle bending constant
+ */
+inline void calc_angle_force(double force1[3], double force2[3],
```

Comment:
please use ``Vector3d``

Reply:
I can use `Vector3d` here and inside the calling functions too for intermediate variables (the return variables are arrays for forces_inline.hpp). At some point in the future we should probably also use `Vector3d` in `/src/core/forces_inline.hpp`.